.github/workflows/rabbit_consumer_schedule.yaml

```diff
@@ -0,0 +1,120 @@
+name: Rabbit Consumer Schedule
+
+# Incrementing the patch version on a schedule to pull latest containers dependencies.
+
+
+on:
+  schedule: 
+      - cron: "12 10 * * TUE" # Every Tuesday at 10.12am
+
+jobs:
+  test_and_lint:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ["3.10"]
+    steps:
+      - uses: actions/checkout@v4
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v5
+        with:
+          python-version: ${{ matrix.python-version }}
+          cache: "pip"
+      - name: Install dependencies
+        run: |
+          sudo apt-get update --fix-missing
+          python -m pip install --upgrade pip
+          # Required for requests-kerberos
+          sudo DEBIAN_FRONTEND=noninteractive apt-get -yq install libkrb5-dev
+          pip install -r OpenStack-Rabbit-Consumer/requirements.txt
+          pip install -r OpenStack-Rabbit-Consumer/requirements-test.txt
+      
+      - name: Run tests
+        # Using Python3 to launch the module sets up the Python path for us
+        run: cd OpenStack-Rabbit-Consumer && python3 -m coverage run -m pytest .
+      
+      - name: Analyse with pylint
+        run: |
+          cd OpenStack-Rabbit-Consumer && pylint $(git ls-files '*.py')
+
+      - name: Prepare coverage
+        run: |
+          cd OpenStack-Rabbit-Consumer && python -m coverage xml
+
+  push_dev_image_harbor:
+    runs-on: ubuntu-latest
+    needs: test_and_lint
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+
+      - name: Login to Harbor
+        uses: docker/login-action@v2
+        with:
+          registry: harbor.stfc.ac.uk
+          username: ${{ secrets.HARBOR_USERNAME }}
+          password: ${{ secrets.HARBOR_TOKEN }}
+
+      - name: Set commit SHA for later
+        id: commit_sha
+        run: echo "sha_short=$(git rev-parse --short HEAD)" >> $GITHUB_OUTPUT
+
+      - name: Build and push to staging project
+        uses: docker/build-push-action@v3
+        with:
+          cache-from: type=gha
+          cache-to: type=gha,mode=max
+          push: true
+          context: "{{defaultContext}}:OpenStack-Rabbit-Consumer"
+          tags: "harbor.stfc.ac.uk/stfc-cloud-staging/openstack-rabbit-consumer:${{ steps.commit_sha.outputs.sha_short }}"
+
+      - name: Inform of tagged name
+        run: echo "Image published to harbor.stfc.ac.uk/stfc-cloud-staging/openstack-rabbit-consumer:${{ steps.commit_sha.outputs.sha_short }}"
```

Comment:
As this runs against `main` we should drop pushing the dev image, as it shouldn't fire

Reply:
(Can be done as a follow-up tidying PR)

Reply:
Roger that, I can clean that up today :+1: